packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -730,14 +753,23 @@ Future<DateTime> showDatePicker({
   @required BuildContext context,
   @required DateTime initialDate,
   @required DateTime firstDate,
-  @required DateTime lastDate
+  @required DateTime lastDate,
+  SelectableDayPredicate selectableDayPredicate
 }) async {
+  assert(!initialDate.isBefore(firstDate), 'initialDate must be on or after firstDate');
```

Comment:
Oh nice.  I guess assert messages work now.  :)

We should start using them in more places.